app/models/ordered_ancestry_entry.rb

```diff
@@ -0,0 +1,230 @@
+    # @param [Hash] nodes_to_arrange_hash - a Hash of nodes to arrange. Initialized to Ancestry.arrange(options) if not given.
+    #
+    # @return [Array<OrderedListEntry>] - an Array with the given node first, then its descendents sorted based on options[:order]
+    #
+    def self.arrange_as_array(options = {}, nodes_to_arrange_hash = nil)
+      nodes_to_arrange_hash ||= arrange(options)
+      arr = []
+      nodes_to_arrange_hash.each do |node, children|
+        arr << node
+        arr += arrange_as_array(options, children) unless children.nil?
+      end
+      arr
+    end
+
+
+    # Inserts the given entry into children before the element with the given +index+.
+    # If no index is given, append the entry at the end. (index = -1)
+    # index is used in the same way as Array#insert
+    # See Array#insert for details
+    #
+    # After the insert, update the order_in_list [= position] for all children.
+    #
+    # @param new_position [Integer] - the 0-based position for the new entry.
+    #  Default = the size of the list of children,  which will place the new_entry at the end
+    # @param new_entry [OrderedListEntry] - the new entry to insert into the list of children
+    #
+    # @return [Array] - children with the new entry inserted
+    def insert(new_entry, new_position = children.size)
+      increment_child_positions(new_position)
+      new_entry.update(list_position: new_position, parent: self)
+    end
+
+
+    alias_method :<<, :insert
+
+
+    # Delete the OrderedListEntry from the list of children.
+    # After the deletion, update the positions for all children as necessary.
+    #
+    # @param entry [OrderedListEntry] - the entry to delete from the list of children
+    # @return [Array] - children with the entry deleted
+    def delete(entry)
+      if children.include?(entry)
+        children.delete(entry)
+        decrement_child_positions(entry.list_position)
+      end
+      children
+    end
+
+
+    # Delete the child at the zero-based position
+    # This must be done BEFORE db changes are written
+    #   else we cannot get the child by list_position because it will have already been changed.
+    #
+    # @return [Array] - children with the entry deleted
+    def delete_child_at(position = children.size)
+      unless position >= children.size
+        deleted_child = child_at_position(position)
+        children.delete(deleted_child)
+        decrement_child_positions(position)
+      end
+      children
+    end
+
+
+    # the next list position that should be used
+    def next_list_position
+      children.size
+    end
+
+
+    def child_at_position(position)
+      return unless children?
+      children.select { |child| child.list_position == position }.first
+    end
+
+
+    # Return a list of all entries that could be parent to this entry.
+    # The entry cannot be a parent to itself
+    # No children of this entry can be a parent
```

Comment:
Nope. It just means that when presenting a list (a select-list drop-down) so that a user can choose which list this is a child of, it only makes sense that a list cannot be a child (sublist) of itself, and it's own children (sublists) cannot be a parent.  (That would be circular. and weird)

You can edit an OrderedListEntry to see how this is put into use.  It's used in the select list (dropdown) where  you can specify the parent of a list:
```
f.collection_select :parent_id, all_allowable_parents, :id, :display_name_with_depth, include_blank: true
```

Here's the index page to see all ordered list entries: `http://0.0.0.0:3000/en/ordered_list_entries`


(rake shf:db_prep; rake db:seed first of course)